src/services/refactors/moveToFile.ts

```diff
@@ -167,6 +170,14 @@ registerRefactor(refactorNameForMoveToFile, {
         if (!interactiveRefactorArguments) {
             return emptyArray;
         }
+        if (context.endPosition !== undefined) {
+            const file = context.file;
+            const startNodeAncestor = getBlockLikeAncestor(getTokenAtPosition(file, context.startPosition));
+            const endNodeAncestor = getBlockLikeAncestor(getTokenAtPosition(file, context.endPosition));
+            if (!isSourceFile(startNodeAncestor) && !isSourceFile(endNodeAncestor)) {
+                return emptyArray;

```

Comment:
It would be good to leave a comment explaining in human terms exactly what the condition is and why it was implemented. I'm pretty sure I suggested this method of implementation, but I already don't remember what it was and it's a little hard to make the leap from code to intuition.